Source/Diagnostics/WarpXOpenPMD.cpp

```diff
@@ -402,6 +403,31 @@ WarpXOpenPMDPlot::~WarpXOpenPMDPlot ()
   }
 }
 
+/*
+ * Flushing out data of the current openPMD iteration
+ * @param [in]: isBTD    if the current diagnostic is BTD
+ *
+ * if isBTD=false, apply the default flush behaviour
+ * if isBTD=true,  advice to use ADIOS Put() instead of PDW for better performance.
+ *
+ *    iteration.seriesFlush() is used instead of series.flush()
+ *    because the latter flushes only if data is dirty
+ *    this causes trouble when the underlying writing function is collective (like PDW)
+ *
+ */
+void WarpXOpenPMDPlot::flushCurrent(bool isBTD) const
+{
+     WARPX_PROFILE("WarpXOpenPMDPlot::flushCurrent");
+
+     openPMD::Iteration currIteration = GetIteration(m_CurrentStep, isBTD);
+     if (isBTD) {
+         currIteration.seriesFlush(  "adios2.engine.preferred_flush_target = \"buffer\"" );
```

Comment:
Need to add a comment:
```suggestion
         // delayed until all fields and particles are registered for flush
         // and dumped once via flattenSteps
         currIteration.seriesFlush(  "adios2.engine.preferred_flush_target = \"buffer\"" );
```